src/frontend/hooks/useTracking.ts

```diff
@@ -1,56 +1,39 @@
-import * as Location from 'expo-location';
-import {useCallback} from 'react';
-import {
-  useTrackActions,
-  useTrackState,
-} from '../contexts/TrackStoreContext.tsx';
-import {LOCATION_TASK_NAME} from '../sharedTypes/location.ts';
-import {useNavigation} from '@react-navigation/native';
-import * as Sentry from '@sentry/react-native';
+import {TrackState, useTrackState} from '../contexts/TrackStoreContext.tsx';
+import {calculateTotalDistance} from '../utils/distance.ts';
 
-export function useTracking() {
-  const {setTracking, clearCurrentTrack} = useTrackActions();
-  const navigation = useNavigation();
-  const isTracking = useTrackState(state => state.isTracking);
-  const locationHistory = useTrackState(state => state.locationHistory);
-
-  const startTracking = useCallback(() => {
-    if (isTracking) {
-      console.warn('Start tracking attempt while tracking already enabled');
-      return;
+type CurrentTrackState =
+  | {
+      hasActiveTrack: false;
+      locationHistory: TrackState['locationHistory'];
+      totalDistance: 0;
+      trackingSince: null;
     }
-    setTracking(true);
-    Location.startLocationUpdatesAsync(LOCATION_TASK_NAME, {
-      accuracy: Location.Accuracy.Highest,
-      activityType: Location.LocationActivityType.Fitness,
-    }).catch(err => {
-      Sentry.captureException(err);
-      setTracking(false);
-      // @ts-expect-error - this is a typing issue, we are using the non-strongly typed hook as this can technically be used in any screen. But regardless of the screen, we want to show the error bottom sheet.
-      navigation.navigate('ErrorBottomSheet');
-    });
-  }, [isTracking, setTracking, navigation]);
-  /**
-   * Cancels location tracking and stops background updates.
-   * @returns {boolean} True if location history has more than one entry (track should be kept), false otherwise.
-   */
-  const cancelTracking = useCallback(() => {
-    setTracking(false);
-    Location.stopLocationUpdatesAsync(LOCATION_TASK_NAME).catch(err => {
-      Sentry.captureException(err);
-    });
-    const hasTrackWithMoreThan1Point = locationHistory.length > 1;
+  | {
+      hasActiveTrack: true;
+      locationHistory: TrackState['locationHistory'];
+      totalDistance: number;
+      trackingSince: Date;
+    };
 
-    if (!hasTrackWithMoreThan1Point) {
-      clearCurrentTrack();
-    }
+export const useCurrentTrackState = (): CurrentTrackState => {
```

Comment:
some hesitation about the `hasActiveTrack` and `trackingSince` being strictly derived from the location history. guess it depends on what we're trying to communicate, but couldn't there be a notable difference between the time when someone enables the tracking and when the first location is recorded? From what I understand, the location history is written to in the background by the background location updater so we don't really have control of when it fires and writes to the location history state.

Maybe the disparity isn't notable in practice, in which case this change is a nice simplification!

Reply:
Thanks for that input! Based on that, it seems like restoring an explicit field for the "tracking since" in the store state seems like the most reasonable option.

Reply:
@gmaclennan I think that true if we use use the [SharedLocationContext](https://github.com/digidem/comapeo-mobile/blob/develop/src/frontend/contexts/SharedLocationContext.tsx) which only updates every 3m.  But if we use expo's [getCurrentPositionAsync](https://docs.expo.dev/versions/latest/sdk/location/#locationgetcurrentpositionasyncoptions), that shouldn't give us the latest and most accurate location, would it not?

I would like to keep it as declarative as possible, so trying to avoid inserting a manual time if possible